cmds/cbmem/main.go

```diff
@@ -0,0 +1,263 @@
+// Copyright 2016-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// cbmem prints out coreboot mem table information in JSON by default,
+// and also implements the basic cbmem -list and -console commands.
+// TODO: checksum tables.
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"reflect"
+)
+
+// The C version of cbmem has a complex function to list
+// numbers in xx,xxx,xxx form. I personally think this
+// is a terrible idea (what about the EU among other things?)
+// If you decide yor really want this, then don't write a function,
+// do this.
+//    "golang.org/x/text/language"
+//    "golang.org/x/text/message"
+//    p := message.NewPrinter(language.English)
+//    p.Printf("%d\n", 1000)
+
+var (
+	mem                 = flag.String("mem", "/dev/mem", "file for coreboot image")
+	debug               = func(string, ...interface{}) {}
+	addr                int64
+	size                int
+	console             bool
+	coverage            bool
+	list                bool
+	hexdump             bool
+	timestamps          bool
+	parseabletimestamps bool
+	verbose             bool
+	version             bool
+	dumpJSON            = flag.Bool("json", true, "Output tables in JSON format")
+)
+
+func init() {
+	flag.BoolVar(&console, "console", false, "print cbmem console")
+	flag.BoolVar(&coverage, "coverage", false, "dump coverage information")
+	flag.BoolVar(&list, "list", false, "print cbmem table of contents")
+	flag.BoolVar(&hexdump, "hexdump", false, "print hexdump of cbmem area")
+	flag.BoolVar(&timestamps, "timestamps", false, "print timestamp information")
+	flag.BoolVar(&parseabletimestamps, "parseable-timestamps", false, "print parseable timestamps")
+	flag.BoolVar(&verbose, "verbose", false, "verbose (debugging) output")
+	flag.BoolVar(&version, "version", false, "print the version")
```

Comment:
Is `version` used?

Reply:
It's in the original and I personally have no idea what we'd use it for. I think it's supposed to indicate the version of the tables you support?

Reply:
I added a version string.